kiota_abstractions/store/backing_store_parse_node_factory.py

```diff
@@ -14,18 +20,12 @@ def __init__(self, concrete: ParseNodeFactory) -> None:
         implementation ParseNodeFactory.
         """
 
-        def func1(x):
-            if isinstance(x, BackedModel):
-                backed_model = x
-                backing_store = backed_model.get_backing_store()
-                if backing_store:
-                    backing_store.set_is_initialization_completed(False)
-
-        def func2(x):
-            if isinstance(x, BackedModel):
-                backed_model = x
-                backing_store = backed_model.get_backing_store()
-                if backing_store:
-                    backing_store.set_is_initialization_completed(True)
-
-        super().__init__(concrete, func1, func2)
+        def on_before_deserialization(x):
+            if isinstance(x, BackedModel) and x.backing_store:
+                x.backing_store.is_initialization_completed = False
```

Comment:
Out of curiousity, should we be calling `set_is_initialization_completed` here as well? Or are they the same thing?

Reply:
`is_initialization_completed` is set to `True` by default, so this changes it to `False` before deserialization.

Reply:
Is that the same as calling `set_is_initialization_completed(False)` as we do in `on_after_deserialization`?

Reply:
Yes